android/src/main/java/com/klippa/reactscanner/KlippaScannerSDKModule.kt

```diff
@@ -310,8 +305,89 @@ class KlippaScannerSDKModule(
                 builder.imageAttributes.storeImagesToGallery = config.getBoolean("StoreImagesToCameraRoll")
             }
 
-            val klippaScanner = builder.build(reactContext)
-            currentActivity.startActivity(klippaScanner)
+            val modes = ArrayList<KlippaDocumentMode>()
+
+            if (config.hasKey("CameraModeSingle")) {
+                val cameraModeSingle = config.getMap("CameraModeSingle") ?: return
+
+                val singleCameraMode = KlippaSingleDocumentMode()
+
+                if (cameraModeSingle.hasKey("name")) {
+                    singleCameraMode.name = cameraModeSingle.getString("name").toString()
+                }
+
+                if (cameraModeSingle.hasKey("message")) {
+                    val message = cameraModeSingle.getString("message").toString()
+
+                    val instructions = Instructions(message = message, dismissHandler = {
+                        singleDocumentModeInstructionsDismissed = true
+                    })
+
+                    singleCameraMode.instructions = instructions
+                }
+
+                modes.add(singleCameraMode)
+            }
+
+            if (config.hasKey("CameraModeMulti")) {
+                val cameraMode = config.getMap("CameraModeMulti") ?: return
+
+                val multipleCameraMode = KlippaMultipleDocumentMode()
+
+                if (cameraMode.hasKey("name")) {
+                    multipleCameraMode.name = cameraMode.getString("name").toString()
+                }
+
+                if (cameraMode.hasKey("message")) {
+                    val message = cameraMode.getString("message").toString()
+
+                    val instructions = Instructions(message = message, dismissHandler = {
+                        multiDocumentModeInstructionsDismissed = true
+                    })
+
+                    multipleCameraMode.instructions = instructions
+                }
+
+                modes.add(multipleCameraMode)
+            }
+
+            if (config.hasKey("CameraModeSegmented")) {
+                val cameraMode = config.getMap("CameraModeSegmented") ?: return
+
+                val segmentedCameraMode = KlippaSegmentedDocumentMode()
+
+                if (cameraMode.hasKey("name")) {
+                    segmentedCameraMode.name = cameraMode.getString("name").toString()
+                }
+
+                if (cameraMode.hasKey("message")) {
+                    val message = cameraMode.getString("message").toString()
+
+                    val instructions = Instructions(message = message, dismissHandler = {
+                        segmentedDocumentModeInstructionsDismissed = true
+                    })
+
+                    segmentedCameraMode.instructions = instructions
+                }
+
+                modes.add(segmentedCameraMode)
+            }
+
+            if (modes.isEmpty()) { return }
```

Comment:
I don't think we should return here, if we return then the scanner will never be launched if the user didn't provide a camera mode. If the user did not provide any camera mode we want the SDK to just use the default values, this can be accomplished like this:

```
if (modes.isNotEmpty()) {
    var index = 0
    if (config.hasKey("StartingIndex")) {
        index = config.getInt("StartingIndex")
    }

     val cameraModes = KlippaCameraModes(
          modes = modes,
          startingIndex = index
     )
     builder.cameraModes = cameraModes
}
```

If there are ever times we need to return early we should also provide the user with a `reject` reason.